File: csharpServer/persistance/IOfficeRepository.cs
using System;
using Avalonia.Data;
using model;

namespace persistance;

public interface IOfficeRepository : IRepository<int, Office>
{
    /// <summary>
    /// Finds an office by its name.
    /// </summary>
    /// <param name="name">The name of the office to be returned.</param>
    /// <returns>An <see cref="Optional{T}"/> encapsulating the office with the given name.</returns>
    /// <exception cref="ArgumentException">Thrown if <paramref name="name"/> is null.</exception>
    Optional<Office> FindByName(string name);
}

File: javaClient/src/main/java/ro/mpp2024/repository/IOfficeRepository.java
package ro.mpp2024.repository;

import ro.mpp2024.model.Office;

import java.util.Optional;

public interface IOfficeRepository extends IRepository<Integer, Office> {

    /**
     *
     * @param name - the name of the office to be returned
     * @return an {@code Optional} encapsulating the office with the given name
     * @throws IllegalArgumentException if name is null
     */
    Optional<Office> findByName(String name);
}
